pkg/cli/cli.go

```diff
@@ -145,8 +146,16 @@ func (c *CLI) buildCmd() error {
 	c.cmd = c.newRootCmd()
 
 	// Get project version and plugin keys.
-	if err := c.getInfo(); err != nil {
-		return err
+	err := c.getInfo()
+	if err != nil {
+		if le, ok := err.(store.LoadError); ok {
+			// ignore UnsupportedVersionError, I'd prefer to verify a specific type
+			// of error instead of the string, but alas LoadError consists wrapped
+			// errors.
```

Comment:
As you say here, this should be checked using a typed error and not an string. We already have a suitable typed error: https://github.com/kubernetes-sigs/kubebuilder/blob/84f357b50e0b25e64c1e5918089f6588b79b81f1/pkg/config/errors.go#L26-L28

So, IMO, changes should be done as follows:

https://github.com/kubernetes-sigs/kubebuilder/blob/84f357b50e0b25e64c1e5918089f6588b79b81f1/pkg/config/store/yaml/store.go#L93
should be
```go
return store.LoadError{Err: config.UnsupportedVersionError{Version: versioned.Version}} 
```

and the check should use `errors.Is` or `errors.As`, not a type assertion. By the way, I would also prefer a:
```go
switch err := c.getInfo(); {
case err == nil:
[...]
}
```
instead of nested if conditionals.